server/fetchers/github_fetcher.go

```
@@ -0,0 +1,184 @@
+package fetchers
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/logging"
+)
+
+type GithubFetcherConfig struct {
+	Repo     string
+	Username string
+	Path     string
+	Token    string
+}
+
+const GithubAPIURLFormat = "https://api.github.com/repos/%s/%s/contents/%s"
+
+const (
+	validObject           string = "0 valid object"
+	nilObjectCode         string = "1 nil object"
+	invalidGithubUserName string = "2 invalid github username"
+	invalidGithubRepoName string = "3 invalid github repo name"
+	invalidGithubPath     string = "4 invalid github path"
+	invalidGithubToken    string = "5 invalid github token"
+)
+
+func (c *GithubFetcherConfig) FetchConfig(logSink *logging.SimpleLogger) (configString string, err error) {
+	// Validate the fetcher config before moving forward
+	err, code := ValidateGithubFetcherConfig(c)
+	if err != nil || code != validObject {
+		logSink.Err("invalid github fetcher config: %s", err)
```

This should just return a wrapped error, rather than logging and then returning the un-wrapped error. Same applies throughout this func.